### ISSUE TITLE
rustup failed to proxy rust-analyzer

### ISSUE THREAD

**Issue by violin0622** [rust-lang/rustup#3237]
### Problem

I'm sure I've installed rust-analyzer, but I can't execute it through rustup proxy.

```
❯ rust-analyzer --version
error: unknown proxy name: 'rust-analyzer'; valid proxy names are 'rustc', 'rustdoc', 'cargo', 'rust-lldb', 'rust-gdb', 'rust-gdbgui', 'rls', 'cargo-clippy', 'clippy-driver', 'cargo-miri', 'rustfmt', 'cargo-fmt'
```

But I can exec it by command below:

```
❯ rustup run stable rust-analyzer --version
rust-analyzer 1.67.1 (d5a82bb 2023-02-07)
```

and other tool are ok:

```
❯ cargo version
cargo 1.67.1 (8ecd4f20a 2023-01-10)

❯ rustc --version
rustc 1.67.1 (d5a82bbd2 2023-02-07)
```

### Steps

1. install rustup
2. install rust-analyzer, cargo, rustc, rust-src etc.
3. try run rust-analyzer
4. error occurs. 

### Notes

OS: OpenSUSE Tumbleweed 20230222

```
❯ uname -a
Linux localhost.localdomain 6.1.12-1-default #1 SMP PREEMPT_DYNAMIC Wed Feb 15 05:31:41 UTC 2023 (373f017) x86_64 x86_64 x86_64 GNU/Linux
```

### Rustup version

```console
❯ rustup --version
rustup 1.25.2 (2023-02-09)
info: This is the version for the rustup toolchain manager, not the rustc compiler.
info: The currently active `rustc` version is `rustc 1.67.1 (d5a82bbd2 2023-02-07)`
```

### Installed toolchains

```console
❯ rustup show
Default host: x86_64-unknown-linux-gnu
rustup home:  /home/violin/.rustup

stable-x86_64-unknown-linux-gnu (default)
rustc 1.67.1 (d5a82bbd2 2023-02-07)
```

**Comment by 0xPoe** [comment]
Duplicate of #3062
We need to wait for a release. I hope we can make it recently.

**Comment by violin0622** [comment]
So, is there an expected release date?
If not, what can we do before that? 
I'm using coc-rust-analyzer which relies on `rust-analyzer` executable.

**Comment by itorres** [comment]
@violin0622 rustup 1.26.0 is in beta and you can test it with:

```
$ export RUSTUP_UPDATE_ROOT=https://dev-static.rust-lang.org/rustup
$ rustup self update
```

Source: https://internals.rust-lang.org/t/help-verify-rustup-1-26-0-release/18577

Also it is quite frustrating that @rbtcollins closed #2411 blaming the end users when the rust-analyzer proxy is not in the current available version and definitely was not three weeks ago.

**Comment by rbtcollins** [comment]
@itorres Thank you for providing the workaround.

On the other aspect - if you want to discuss that, please join us in zulip or discord and we can do so. I can see that I made a mistake in what I said - I honestly did think we'd shipped it after the urgent release not so long ago - but I stand by the decision to close and lock the bug. This is a bugtracker, a tool for use by the volunteers doing the work, and when bugs become hard to read they stop serving their purpose. 

What we need is folk working with the people filing bugs to figure out how to solve them, and then putting patches together to solve them. Some things are simple, some are not, and so we need a place for discussions about the approach to take place (see for instance the request for a hidden toolchain file for the linux kernel folk), and we've got a very small number of folk contributing here (historically the case, not just now, not as a result of bug management as far as I can tell).

Anyhow - follow-up-to zulip/discord please.
